apps/vessel-history/features/map/map-events.hooks.ts

```diff
@@ -32,14 +31,19 @@ export default function useMapEvents() {
   const { viewport, setMapCoordinates } = useViewport()
   const [findEventVoyage, setFindEventVoyage] = useState<RenderedEvent>()
   const { dispatchQueryParams } = useLocationConnect()
+  const [cursorLayers, setCursorLayers] = useState<ExtendedFeature[] | null>()
 
   const selectVesselEventOnClick = useCallback(
     (event: InteractionEvent | null) => {
       const features = event?.features ?? []
-
       const vesselFeature = features.find(
         (feature) => feature.generatorType === GeneratorType.VesselEvents
       )
+      setCursorLayers(
```

Comment:
why not store all the features and render the content based on the generatorType in the [PopupWrapper](https://github.com/GlobalFishingWatch/frontend/blob/develop/apps/fishing-map/features/map/popups/PopupWrapper.tsx#L140) this was we are flexible for different layers in the future